src/main/scala/io/qbeast/core/model/QbeastSnapshot.scala

```diff
@@ -24,12 +26,26 @@ import org.apache.spark.sql.Dataset
  */
 trait QbeastSnapshot {
 
+  /**
+   * Returns the current snapshot object, representing the dataset state.
+   */
+  def snapshot: Any
```

Comment:
I don't like this returning an any. Do we need to return the underlying snapshot type? I think it is best if we don't declare this method in the trait. 
On the hand, if we really have to do it, it is best to keep it as generic 
```
trait QbeastSnapstho[T] {
def snapshot:T
```

But this makes only sense if we are confident we have similar "snapshot" objects in all 3 open table formats.